update/v1alpha1/types_clusterversion_progress_insight.go

```diff
@@ -0,0 +1,263 @@
+type ClusterVersionProgressInsightConditionType string
+
+const (
+	// Updating condition communicates whether the ClusterVersion is updating
+	ClusterVersionProgressInsightUpdating ClusterVersionProgressInsightConditionType = "Updating"
+)
+
+// ClusterVersionProgressInsightUpdatingReason are well-known reasons for the Updating condition on ClusterVersion status insights
+type ClusterVersionProgressInsightUpdatingReason string
+
+const (
+	// CannotDetermineUpdating is used with Updating=Unknown
+	ClusterVersionCannotDetermineUpdating ClusterVersionProgressInsightUpdatingReason = "CannotDetermineUpdating"
+	// ClusterVersionProgressing means that ClusterVersion is considered to be Updating=True because it has a Progressing=True condition
+	ClusterVersionProgressing ClusterVersionProgressInsightUpdatingReason = "ClusterVersionProgressing"
+	// ClusterVersionNotProgressing means that ClusterVersion is considered to be Updating=False because it has a Progressing=False condition
+	ClusterVersionNotProgressing ClusterVersionProgressInsightUpdatingReason = "ClusterVersionNotProgressing"
+)
+
+// ResourceRef is a reference to a kubernetes resource, typically involved in an insight
+type ResourceRef struct {
+	// group of the object being referenced, if any
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	Group string `json:"group,omitempty"`
+
+	// resource of object being referenced
+	// +required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	Resource string `json:"resource"`
+
+	// name of the object being referenced
+	// +required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	Name string `json:"name"`
+
+	// namespace of the object being referenced, if any
+	// +optional
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// ScopeType is one of ControlPlane or WorkerPool
+// +kubebuilder:validation:Enum=ControlPlane;WorkerPool
+type ScopeType string
```

Comment:
nit: `ScopeType` doesn't seemed to be consumed by `ClusterVersionProgressInsight`.  Maybe move it to `types_health_insight.go` to live alongside the consuming structure?

Reply:
It's also used in MCP and Node insights. Introducing a "common.go" felt like overhead but maybe it's worth it (especially if we keep using `ResourceRef` everywhere and not just in Health)

Reply:
Moved to `types_common.go`